calculator_test.go

```diff
@@ -2,23 +2,213 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
+	tcs := []struct {
+		a, b float64
+		rest []float64
+		want float64
+		name string
+	}{
+		{name: "Add two units", a: 0, b: 0, rest: []float64{}, want: 0},
+		{name: "Add two nonunits", a: 1, b: 2, rest: []float64{}, want: 3},
+		{name: "Add unit and nonunit", a: 0, b: 2, rest: []float64{}, want: 2},
+		{name: "Add negative and inverse", a: -2, b: 2, rest: []float64{}, want: 0},
+		{name: "Add negative and positive", a: -2, b: 3, rest: []float64{}, want: 1},
+		{name: "Add two numbers", a: 1, b: 2, rest: []float64{}, want: 3},
+		{name: "Add three numbers", a: 1, b: 2, rest: []float64{3}, want: 6},
+		{name: "Add four numbers", a: 1, b: 2, rest: []float64{3, 4}, want: 10},
+		{name: "Add five numbers", a: -1, b: 2, rest: []float64{-3, 4, -5}, want: -3},
+	}
+	for _, tc := range tcs {
+		got := calculator.Add(tc.a, tc.b, tc.rest...)
+		if tc.want != got {
+			t.Errorf("Wanted %v, got %v in %v", tc.want, got, tc.name)
+		}
+	}
+}
+func TestAddRandom(t *testing.T) {
+	t.Parallel()
+	for i := 0; i < 100; i++ {
+		a, b := rand.Float64(), rand.Float64()
+		want := a + b
+		got := calculator.Add(a, b)
+		if want != got {
+			t.Errorf("Want %f, got %f, using %f + %f", want, got, a, b)
+		}
+	}
+}
+
+func TestSubtract(t *testing.T) {
```

Comment:
Since the test case structs and the test logic are identical for Add, Subtract, and Multiply, we could combine all three tests into one, couldn't we?